modules/fbx_importer/data/fbx_mesh_data.cpp

```diff
@@ -320,14 +360,174 @@ MeshInstance *FBXMeshData::create_fbx_mesh(const ImportState &state, const Assim
 		in_mesh_surface_id += 1;
 	}
 
-	// TODO please add skinning.
-	//mesh_id = mesh_geometry->ID();
-
 	MeshInstance *godot_mesh = memnew(MeshInstance);
 	godot_mesh->set_mesh(mesh);
 	return godot_mesh;
 }
 
+void FBXMeshData::reorganize_vertices(
+		std::vector<int> &r_polygon_indices,
+		std::vector<Vector3> &r_vertices,
+		HashMap<int, Vector3> &r_normals,
+		HashMap<int, Vector2> &r_uv_1,
+		HashMap<int, Vector2> &r_uv_2,
+		HashMap<int, Color> &r_color,
+		HashMap<String, MorphVertexData> &r_morphs,
+		HashMap<int, HashMap<int, Vector2> > &r_uv_1_raw,
+		HashMap<int, HashMap<int, Vector2> > &r_uv_2_raw) {
+
+	// Key: OldVertex; Value: [New vertices];
+	HashMap<int, Vector<int> > duplicated_vertices;
+
+	PolygonId polygon_index = -1;
+	for (int pv = 0; pv < (int)r_polygon_indices.size(); pv += 1) {
+		if (is_start_of_polygon(r_polygon_indices, pv)) {
+			polygon_index += 1;
+		}
+		const Vertex index = get_vertex_from_polygon_vertex(r_polygon_indices, pv);
+
+		bool need_duplication = false;
+		Vector2 this_vert_poly_uv1 = Vector2();
+		Vector2 this_vert_poly_uv2 = Vector2();
+
+		HashMap<int, HashMap<int, Vector2> > *uv_raw = &r_uv_1_raw;
+		Vector2 *this_vert_poly_uv = &this_vert_poly_uv1;
+		for (int kk = 0; kk < 2; kk += 1) {
+			if (uv_raw->has(index)) {
+				const HashMap<PolygonId, Vector2> *uvs = uv_raw->getptr(index);
+
+				if (uvs->has(polygon_index)) {
+					// This Polygon has its own uv.
+					(*this_vert_poly_uv) = *uvs->getptr(polygon_index);
+
+					// Check if we need to duplicate it.
+					for (const PolygonId *pid = uvs->next(nullptr); pid != nullptr; pid = uvs->next(pid)) {
+						if (*pid == polygon_index) {
+							continue;
+						}
+						const Vector2 vert_poly_uv = *uvs->getptr(*pid);
+						if (((*this_vert_poly_uv) - vert_poly_uv).length() > CMP_EPSILON) {
+							// Yes this polygon need duplication.
+							need_duplication = true;
+							break;
+						}
+					}
+				} else if (uvs->has(-1)) {
+					// It has the default UV.
+					(*this_vert_poly_uv) = *uvs->getptr(-1);
+				} else if (uvs->size() > 0) {
+					// No uv, this is strange, just take the first and duplicate.
+					(*this_vert_poly_uv) = *uvs->getptr(*uvs->next(nullptr));
+					WARN_PRINT("No UVs for this polygon, while there is no default and some other polygons has it. This FBX file may be corrupted.");
```

Comment:
"No UVs for this polygon, while there is no default and some other polygons has it. This FBX file may be corrupted."

This sentence is awkward.

Reply:
Fixed but I'm still not happy with the fix I did.